File: 134-gas-station/gas-station.js
/**
 * @param {number[]} gas
 * @param {number[]} cost
 * @return {number}
 */
var canCompleteCircuit = function(gas, cost) {
    let now = 0;
    let total = 0;
    let idx = 0;
    for(let i = 0; i < gas.length; i++){
        let fuel = gas[i] - cost[i];
        total += fuel;
        now += fuel;
        if(now < 0){
            now = 0;
            idx = i + 1;
        }
    }
    if(total >=0) return idx;
    return -1;
};

File: 134-gas-station/gas-station.java
class Solution {
    public int canCompleteCircuit(int[] gas, int[] cost) {
    int now = 0;
    int total = 0;
    int idx = 0;
    for(int i = 0; i < gas.length; i++){
        int fuel = gas[i] - cost[i];
        total += fuel;
        now += fuel;
        if(now < 0){
            now = 0;
            idx = i + 1;
        }
    }
    if(total >=0) return idx;
    return -1;
    }
}